src/channel.c

```diff
@@ -88,7 +88,8 @@ Cflag *find_cflag_recursive(const Chan *chptr, const User *uptr)
             fprintf(stderr, "Impossible recursive link master=%s, slave=%s\n", l->master, l->slave);
             return NULL;
         }
-        return find_cflag_recursive(chptr, uptr2);
+        cflag = find_cflag_recursive(chptr, uptr2);
+        cflag->user = uptr;
```

Comment:
This is effectively mutating persistent state by changing the account associated with a channel access and will be reflected in the next db write. This should not happen as there is no intent to modify a channel access. Can you clarify the purpose of this change?

Reply:
The purpose here is to make sure `find_cflag_recursive` returns the secondary account name that is then used in https://github.com/xiu/child/blob/75b135b65cf93103530bbcdb12494b6d6a5e9a61/src/parseline.c#L1390. I had not thought about it being persistent though. I'll probably need to revert that change and add a logic to lookup the main account in https://github.com/xiu/child/blob/link_fix/src/user.c#L499 or do you see a better way?

Reply:
One option could be to change the sync_cflag() function by adding an Nick* parameter to which the cflag should apply:
void sync_cflag(const Cflag *cflag, Nick *nptr)
Because sync_cflag always requires a valid nptr, it should not accept a NULL pointer.

In m_sjoin, the function call after find_cflag_recursive would then become: sync_cflag(cflag, nptr), ensuring the access are applied to the correct nick.

This would of course require updating all sync_cflag call sites.